File: Arrays&Hashing/ConcatenationOfArray.py
# https://leetcode.com/problems/concatenation-of-array/description/
def getConcatenation(nums: list[int]) -> list[int]:
    return [*nums,*nums]

print(getConcatenation([1,2,1])) # [1,2,1,1,2,1]
print(getConcatenation([1,3,2,1])) # [1,3,2,1,1,3,2,1]
print(getConcatenation([1])) # [1,1]

File: Arrays&Hashing/ConcatenationOfArray.java
class ConcatenationOfArray {
    public int[] getConcatenation(int[] nums) {
        int[] ans = new int[nums.length*2];
        for (int i = 0; i < nums.length; i++) {
            ans[i] = nums[i];
            ans[i+nums.length] = nums[i];
        }
        return ans;
    }
    public static void main(String[] args) {
        ConcatenationOfArray concat = new ConcatenationOfArray();
        System.out.println(concat.getConcatenation(new int[] {1,2,3}));
    }
}